single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/styles/iidm/TopologicalStyleProvider.java

```diff
@@ -99,27 +101,32 @@ private Map<String, String> createBusIdStyleMap(String baseVoltageLevelStyle, St
         if (svgParameters.isUnifyVoltageLevelColors()) {
             for (int i = 0; i < buses.size(); i++) {
                 Bus bus = buses.get(i);
-                busIdStyleMap.put(bus.getId(), baseVoltageLevelStyle + '-' + i);
+                busIdStyleMap.put(bus.getId(), baseBusStyle + '-' + i);
             }
         } else {
             for (Bus b : buses) {
-                int newIndex = stylesIndices.compute(baseVoltageLevelStyle, (s, i) -> i == null ? 0 : i + 1);
-                String style = baseVoltageLevelStyle + '-' + newIndex;
+                int newIndex = stylesIndices.compute(baseVoltageName, (s, i) -> i == null ? 0 : i + 1);
+                String style = baseBusStyle + '-' + newIndex;
                 busIdStyleMap.put(b.getId(), style);
             }
         }
 
         return busIdStyleMap;
     }
 
-    private Optional<String> getNodeTopologicalStyle(String baseVoltageLevelStyle, String vlId, Node node) {
+    private List<String> getNodeTopologicalStyle(String baseVoltageLevelStyle, String vlId, Node node) {
```

Comment:
Could you add a "s" as it's now a list?
```suggestion
    private List<String> getNodeTopologicalStyles(String baseVoltageName, String vlId, Node node) {
```